pkg/staticnetworkconfig/generator.go

```diff
@@ -90,40 +92,16 @@ func (s *StaticNetworkConfigGenerator) generateHostStaticNetworkConfigData(ctx c
 	return filesList, nil
 }
 
-func (s *StaticNetworkConfigGenerator) executeNMStatectl(ctx context.Context, hostYAML string) (string, error) {
-	err := s.sem.Acquire(ctx, 1)
+func (s *StaticNetworkConfigGenerator) validateNMStateConfig(hostYAML string) (string, error) {
+	hostJSON, err := yamlconvertor.YAMLToJSON([]byte(hostYAML))
 	if err != nil {
-		s.log.WithError(err).Errorf("Failed to lock semaphore for nmstatectl execution")
-		return "", err
+		fmt.Printf("err: %v\n", err)
```

Comment:
I think you should handle this error.

Also remove debugging printf statements.

Reply:
Removed.
returning to error, so it is handled by the calling function